File: src/ValidateBrackets/golang/src/solution/solution_test.go
package solution

import "testing"

func TestIsValid(t *testing.T) {
    tests := []struct {
        input    string
        expected bool
    }{
        {"()", true},
        {"()[]{}", true},
        {"(]", false},
        {"([)]", false},
        {"{[]}", true},
        {"", true}, // Empty string should also return true
    }

    for _, test := range tests {
        if result := IsValid(test.input); result != test.expected {
            t.Errorf("IsValid(%q) = %v; expected %v", test.input, result, test.expected)
        }
    }
}

File: src/ValidateBrackets/java/src/test/java/com/validatebrackets/SolutionTest.java
package com.validatebrackets;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import org.junit.Test;

public class SolutionTest {
    @Test
    public void testIsValid() {
        Solution solution = new Solution();
        
        // Test case 1
        assertTrue("Expected true for empty string", solution.isValid(""));
        // Test case 2
        assertTrue("Expected true for ()", solution.isValid("()"));
        // Test case 3
        assertTrue("Expected true for ()[]{}", solution.isValid("()[]{}"));
        // Test case 4
        assertFalse("Expected false for (]", solution.isValid("(]"));
        // Test case 5
        assertFalse("Expected false for ([)]", solution.isValid("([)]"));
        // Test case 6
        assertTrue("Expected true for {[]}", solution.isValid("{[]}"));
    }
}

